### DISCUSSION TITLE
Strange delay to download a file

### DISCUSSION THREAD

**Discussion by vincentdev15** [livewire/livewire#5012] [1. Help]
Hello,

I just compared two ways to download a file.

Via the Livewire controller, it takes 2 or 3 seconds to download the file, I mean the download begins 2 or 3 seconds after the click on the button.

Via a standard controller, the download begins immediately.

The same code is used in both cases.

return response()->download(storage_path('backups/').$filename);

I don't understand why it takes more time to download the same file via Livewire.

Do you have any idea ?

Thanks.

Vincent

**Comment by joshhanley** [comment]
@vincentdev15 Livewire's download is triggered from the frontend, so Livewire needs to finish its request first, send the file to the browser, then trigger the download. So that's probably why you are seeing the delay. 

Try adding in a `wire:loading` element someone and see if that is still loading during the delay.

Hope this helps!
